sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c

```diff
@@ -774,21 +868,28 @@ static int create_sdw_codec_dai(struct device *dev,
 		if (!codec_info_list[idx].direction[i])
 			continue;
 
+		/* create stream name based on the first link id */
 		name = devm_kasprintf(dev, GFP_KERNEL,
-				      sdw_stream_name[i], link_id);
+				      sdw_stream_name[i], cpu_dai_id[0]);
 		if (!name)
 			return -ENOMEM;
 
-		cpu_name = devm_kasprintf(dev, GFP_KERNEL,
-					  "SDW%d Pin%d", link_id, j + 2);
-		if (!cpu_name)
-			return -ENOMEM;
+		for (k = 0; k < cpu_dai_num; k++) {
+			cpu_name = devm_kasprintf(dev, GFP_KERNEL,
+						  "SDW%d Pin%d", cpu_dai_id[k],
+						  j + 2);
+			if (!cpu_name)
+				return -ENOMEM;
+
+			cpus[cpu_dai_index++].dai_name = cpu_name;
```

Comment:
this might be correct but this gives me an uncomfortable feeling in my stomach :-) This assumes that `cpus[]` is big enough to hold `cpu_dai_num` additional elements, and `cpu_dai_num` was calculated in `get_slave_info()` and `cpus[]` was allocated before that in `sof_card_dai_links_create()` and that one used a different method - `get_sdw_dai_info()` to get all the numbers... So, you might understand a moderately diligent reader for having a difficulty verifying that :-) Maybe at least add a check that we aren't overrunning the array, but in fact I'd say, we need to avoid re-calculating those numbers, if possible. Assuming `cpu_dai_num` in both cases should be the same number?

Reply:
thanks, I will add check for it.